### DISCUSSION TITLE
Ability to monitor Cloudshell with Azure Policy (vnet, storage, logging, RBAC)

### DISCUSSION THREAD

**Discussion by jdelabr** [Azure/CloudShell#270] [Ideas]
Dear all,

We need a way to be able to audit and enforce the following:
- VNet integration
- Use of compliant Storage account (with logging, no exposure to internet and CMK)
- Log all commands passed onto cloudshell

In addition we would require the ability to manage and audit privileges to use cloudshell at Azure RBAC level.

Thank you in advance!

Kind regards,
Julien

**Comment by maertendMSFT** [comment]
Would folks prefer these types of controls at a subscription or tenant level?

**Reply by jdelabr** [reply]
@maertendMSFT 
Those would be best managed via Azure policy in my opinion, which I understand can be applied to subscriptions and/or management groups.

**Reply by kirkwhetton** [reply]
Either of those would be preferable to nothing at all! Upvoted.

**Reply by chrisking81** [reply]
My FSI customer needs to fully disable Cloud Shell, as it provides a container running in an unfiltered vNet, with direct access to the internet.

Disabling for the whole tenant would be the best option, as I assume it would also disable the ability to mount the Cloud Shell in VS Code and Windows Terminal as well.

Allowing a potential special group of users (Cloud Ops) to use under breakglass/PIM elevation could be a bonus.

**Reply by AndFrantz** [reply]
this would be a big win for me! 

My vote would be for subscription. We may need to granularly control who can do cloud shell based on risk or application tier. 

Ideally we would also be able to restrict Non-VNET isolated cloud shell, while allowing customers to use managed in-VNET cloud shell. This will be a common requirement for regulated networks where egress traffic must go through a network security choke point.

**Comment by MiloCoding123** [comment]
Does anyone have a script that automatically creates and monitors breakglass accounts? I'm trying to make one with the new AzureAD module but if anyone already has one it will be greatly appreciated! Its on powershell
